plugins/flytekit-data-fsspec/tests/test_persist.py

```diff
@@ -18,6 +18,17 @@ def test_s3_setup_args():
     kwargs = s3_setup_args(S3Config(access_key_id="access"))
     assert kwargs == {"key": "access"}
 
+    os.environ["AWS_ACCESS_KEY_ID"] = "ignore-user"
```

Comment:
Could we use this approach for handling the env vars https://stackoverflow.com/questions/31582750/python-mock-patch-os-environ-and-return-value

and also add a few more tests to cover at least the three interesting possibilities?
1. Std env vars present, `FLYTE_` env vars not present
2. Std env vars present, `FLYTE_` env vars also present
3. Std env vars not present, `FLYTE_` env vars present

Reply:
Could you also run a local test using pyflyte run for sanity check please?

1. Build an image with this new flytekit code and push the image to the sandbox registry.
    a. Basically on line [22 in the dockerfile](https://github.com/flyteorg/flytekit/blob/be24c52f3fb8f7949db172f490cca95fa0f0e413/Dockerfile#L22) add a pip command to install this git sha of flytekit.
        It'll look something like
            ```RUN pip install --no-deps -U "https://github.com/flyteorg/flytekit/archive/abcabc.zip#egg=flytekit"
            ```
    b. `docker build . --tag localhost:30000/test_stuff:v1 --build-arg=PYTHON_VERSION=3.10 --build-arg=VERSION=1.4.1`
    c. `docker push localhost:30000/test_stuff:v1`
1. Then `pyflyte run --remote --image localhost:30000/test_stuff:v1 some/workflow.py`

Somewhat annoying I know, but a simple test like this would've prevented the 1.4.0 regression we had earlier today.  just trying to be more careful and we'll work on making this easier/automated.

Reply:
updated the tests.

Reply:
```
RUN pip install --no-deps -U "https://github.com/flyteorg/flytekit/archive/291f9aeb520e8fd1ca6837a6ea78f6722a3d3d16.zip#subdirectory=plugins/flytekit-data-fsspec&egg=flytekitplugins-data-fsspec"
```

Reply:
ok, I can confirm that this works as expected in the sandbox